src/main/java/org/apache/mesos/scheduler/DefaultScheduler.java

```diff
@@ -81,63 +79,67 @@ void awaitTermination() throws InterruptedException {
     }
 
     Plan getPlan() {
-        return plan;
+        return deployPlan;
     }
 
     private void initialize(SchedulerDriver driver) throws InterruptedException {
-        logger.info("Initializing.");
+        logger.info("Initializing...");
         initializeGlobals(driver);
-        initializeRecoveryScheduler();
+        initializeRecoveryPlanManager();
         initializeDeploymentPlan();
         initializeResources();
+        final List<PlanManager> planManagers = Arrays.asList(
+                deployPlanManager,
+                recoveryPlanManager);
+        planCoordinator = new DefaultPlanCoordinator(planManagers, planScheduler);
+        logger.info("Done initializing.");
     }
 
     private void initializeGlobals(SchedulerDriver driver) {
-        logger.info("Initializing globals");
+        logger.info("Initializing globals...");
         stateStore = new CuratorStateStore(serviceSpecification.getName(), zkConnectionString);
         taskFailureListener = new DefaultTaskFailureListener(stateStore);
         taskKiller = new DefaultTaskKiller(stateStore, taskFailureListener, driver);
         reconciler = new DefaultReconciler(stateStore);
         offerAccepter = new OfferAccepter(Arrays.asList(new PersistentOperationRecorder(stateStore)));
+        planScheduler = new DefaultPlanScheduler(offerAccepter, new OfferEvaluator(stateStore), taskKiller);
     }
 
-    private void initializeRecoveryScheduler() {
-        logger.info("Initializing recovery scheduler");
-        RecoveryRequirementProvider recoveryRequirementProvider =
+    private void initializeRecoveryPlanManager() {
+        logger.info("Initializing recovery plan...");
+        final RecoveryRequirementProvider recoveryRequirementProvider =
                 new DefaultRecoveryRequirementProvider(new DefaultOfferRequirementProvider());
-        LaunchConstrainer constrainer =
+        final LaunchConstrainer constrainer =
                 new TimedLaunchConstrainer(Duration.ofSeconds(DELAY_BETWEEN_DESTRUCTIVE_RECOVERIES_SEC));
-        recoveryScheduler = new DefaultRecoveryScheduler(
+
+        recoveryPlanManager = new SimpleRecoveryPlanManager(
                 stateStore,
                 taskFailureListener,
                 recoveryRequirementProvider,
-                offerAccepter,
                 constrainer,
-                new TimedFailureMonitor(Duration.ofSeconds(PERMANENT_FAILURE_DELAY_SEC)),
-                recoveryStatusRef);
+                new TimedFailureMonitor(Duration.ofSeconds(PERMANENT_FAILURE_DELAY_SEC)));
     }
 
     private void initializeDeploymentPlan() {
```

Comment:
This initializes both the `DeploymentPlan` and the `DeploymentPlanManager`.  Maybe they should be separated or the name should be changed.  It also makes `initialize()` confusing because we intialize the recovery manager, but just the deployment plan.  I think breaking it up a bit makes the initialize method clearer, but up to you.